oeps/processes/oep-0055-proc-project-maintainers.rst

```diff
@@ -0,0 +1,399 @@
+
+Historical Context
+------------------
+Historically, the Open edX project has not had a well-defined role
+of maintainer.  At times, community members stepped into this role
+informally.  The role was also part of a broader "owner" role at edX.
+Owners were also responsible for the health of the component they
+owned both in the repository and in production; owners answered the
+pager when something went wrong.
+
+What Maintainers Do
+-------------------
+The role of the repository or project maintainer is both a community
+focused and a technical role[#]_.  Maintainers will be the public face
+of each component and should be welcoming and available.  While
+maintainers should strive to always be friendly and kind in their
+communication, firmness and directness are equally important.
+Maintainers must enforce the project's standards.  In cases where a
+proposed change isn't aligned with the our standards or direction, it
+is best to make that clear as early as possible.
+
+Non-technical responsibilities of maintainers include:
+
+* Effective communication with community members, other maintainers,
+  and project leadership
+* Selection and approval of Core Contributors with commit rights to their repositories
+* Disciplined management to ensure that maintenance work is done
+* Enforcement of the project's standards
+* The ability to solve or escalate issues facing users of the
+  component
+
+Technical responsibilities of maintainers include:
+
+* Understanding the current component architecture and implementation
+* Ensuring that basic documenation for the component exists and is
+  accurate
+* Having a documented technical vision for the evolution of the
+  component
+* The ability to assess the appropriateness of proposed changes to or
+  extensions of the component
+* The role of the component in the overall platform and how it
+  interacts with other components
+
+It is acceptable for a component to be maintained by a team or group
+as long as that arrangement does not reduce the reachablility of the
+maintainer or reduce accountability.
+
+Prerequisites
+-------------
+In order to be a repository maintainer, you must
+
+* Be an active member of the Open edX community who has made technical
+  contributions to the project in the last 6 months.  The types of
+  contributions that are relevant will match those defined in the
+  `code contributors`_
+* Abide by the `Open edX Code of Conduct
+  <https://openedx.org/code-of-conduct/>`_
+* Be a core contributor for the repository in question who has
+  demonstrated excellence in the role of core contributor.  From
+  time-to-time, in the case of a strong candidate and project need,
+  core contributor status and maintainer status may be conferred at
+  the same time.
+
+Bootstrapping
+-------------
+In order to ensure an orderly roll-out of the formal maintainer role,
+existing 2U staff in the owner role will transition into the role of
```

Comment:
I'm wondering aloud if we should make this a more explicit choice? That is, the current informal owners (both 2U staff and other community members) can:

1. Give up maintainer-ship of the component, so that someone else can pick it up
OR
2. Bring the component up to standard within 1 month

We might also consider giving an extra long grace-period for bringing components up to standard as part of the bootstrapping process in order to give the prior informal owners time to prioritize the needed documentation work.

Reply:
I think this makes sense generally.  However, if a large number of components were put up for adoption that would not lead to an orderly transition.  I would say:

1. If there are components that are unloved, but owned, I think the current maintainer should see them through the DEPR process.
2. If there are components that are not a high priority for a given team, we should have a process for seeking a new maintainer, before they are "given up."

Fully agree that we should have a defined timeframe for conformance.  However, I also don't expect us to add a ton of new requirements now.

Reply:
I think things like rollout plans and conformance timelines should appear in other documents (probably wiki pages, possibly ADRs in certain repos) rather than codified in an OEP. OEPs aren't meant to change super frequently, and we'd want to keep timelines and execution plans more agile.

Reply:
Do we have a census of the repos/components and how well they meet the requirements here, or a tool to validate a repo?

The repo health spreadsheet seems to check for some of these along with a bunch of others, so it was hard for me to tell how well a given repo meets this standard.

Reply:
Also, are there recommended practices for these standards?  E.g. how to keep the changelog up-to-date and what level of detail belongs there, recommended answers for how to report issues and where to go for help, etc.

Reply:
@e0d I recommend we remove this "Bootstrapping" section into a wiki page. That wiki page can be linked in this OEP, but debating and iterating on the specifics of the pilot and subsequent phases doesn't need to take place in this PR.